### ISSUE TITLE
`whyNoLint` does not work as expected if {linter} name is specified in `nolint` inline directive

### ISSUE THREAD

**Issue by niya3** [go-critic/go-critic#1063]
i'm expecting `whyNotLint` working both with `//nolint //{comment}` and `//nolint: {liter} //{comment}` forms, but it doesn't happen:

```bash
$ golangci-lint version    
golangci-lint has version v1.41.0 built from (unknown, mod sum: "h1:1kfytWKEn2EPjniwD5xCwWtHAHR57q+vmDX3+VBoJac=") on (unknown)
$ golangci-lint run main.go
main.go:13:37: whyNoLint: include an explanation for nolint directive (gocritic)
        ErrDHCP = errors.New("DHCP error") //nolint: stylecheck // i'm ok with it too
```

```go
// main.go file
package main

import (
	"errors"
	"fmt"
)

var (
	ErrDNS = errors.New("DNS error") //nolint // actually hides warning
)

var (
	ErrDHCP = errors.New("DHCP error") //nolint: stylecheck // does not hide warning
)

func main() {
	fmt.Println("vim-go")
}
```

```yaml
// .golangci.yml file
linters:
  disable-all: true
  enable:
    - gocritic
    - stylecheck

linters-settings:
  gocritic:
    enabled-tags:
      - style
      - experimental
```

**Comment by cristaloleg** [comment]
Isn't this the same? https://github.com/go-critic/go-critic/issues/1050

**Comment by niya3** [comment]
@cristaloleg yep, it is, thx! fixed by removing additional spaces, i.e. replacing everywhere

```diff
-// nolint: {linter} // {comment}
+//nolint:{linter} // {comment}
```
